teps/0114-custom-tasks-beta.md

```diff
@@ -0,0 +1,293 @@
+---
+status: implementable
+title: Custom Tasks Beta
+creation-date: '2022-07-12'
+last-updated: '2022-07-12'
+authors:
+- '@jerop'
+see-also:
+- TEP-0002
+- TEP-0061
+- TEP-0069
+- TEP-0071
+- TEP-0096
+- TEP-0105 
+---
+
+# TEP-0114: Custom Tasks Beta
+
+<!-- toc -->
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+  - [Proposal](#proposal)
+    - [Required](#required)
+      - [API Changes](#api-changes)
+        - [v1alpha1 to v1beta1 + Run to CustomRun](#v1alpha1-to-v1beta1--run-to-customrun)
+        - [Embedded Specifications](#embedded-specifications)
+        - [Feature Gates](#feature-gates)
+      - [Documentation](#documentation)
+      - [Testing](#testing)
+    - [Optional](#optional)
+      - [Authoring Custom Tasks](#authoring-custom-tasks)
+      - [Sharing Custom Tasks](#sharing-custom-tasks)
+        - [Custom Tasks in Tekton Hub](#custom-tasks-in-tekton-hub)
+  - [References](#references)
+<!-- /toc -->
+
+## Summary
+
+The aim of this TEP is to promote `Custom Tasks`, which are used to extensibility in Tekton Pipelines, from alpha to 
+beta.
```

Comment:
I have a general question, not necessarily just applies to this TEP. 

@vdemeester @afrittoli when we release `V1`, would the custom tasks be released as `V1` or stays back in `v1beta1`? It doesn't make sense to keep it in `v1beta1` since its the highest stable version right now but all `beta` features will become `v1` 🤔

Reply:
If by "custom task" you mean the CRD related to CustomTask, aka `CustomRun` (with that TEP) could stay `v1beta1` and be refered by a `v1` API (`PipelineRun` CRD). Each CRD can have their own api lifecycle. And most likely, each should go through `v1alpha1` -> `v1beta1` -> `v1`.